### PR TITLE
retry_cluster_exceptions return real last exception

### PR DESCRIPTION
**Pull request by myakove**
##### Short description:
`retry_cluster_exceptions` will raise last exception from `TimeoutSamler` if any, otherwise `TimeoutExpiredError` will be raised

##### Which issue(s) this PR fixes:
https://github.com/RedHatQE/openshift-python-wrapper/issues/1678

### CODE REVIEW

**Comment by redhat-qe-bot2** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: ocp_resources/resource.py [hunk 1]
```diff
@@ -765,16 +765,22 @@ def update_replace(self, resource_dict):
...
-            func=func,
-            print_log=False,
-            exceptions_dict=exceptions_dict,
-            **kwargs,
-        )
-        for sample in sampler:
-            return sample
+        try:
+            sampler = TimeoutSampler(
+                wait_timeout=10,
```
Comment (jpeimer) [inline ocp_resources/resource.py:770]:
```suggestion
                wait_timeout=TIMEOUT_10SEC
```

**Comment by myakove** [comment]
depends on: https://github.com/RedHatQE/timeout-sampler/pull/34

File: ocp_resources/resource.py (hunk 1 above)
Comment (myakove) [reply ocp_resources/resource.py:770]: ack

**Review by rnetser — APPROVED** [review]

**Comment by redhat-qe-bot2** [comment]
Cherry-picked PR retry_cluster_exceptions return real last exception into v4.15

**Pull request merged**